src/NuGet.Core/NuGet.Packaging/ContentModel/ContentItem.cs

```diff
@@ -7,12 +7,25 @@ namespace NuGet.ContentModel
 {
     public class ContentItem
     {
+        private Dictionary<string, object> _properties;
         public string Path { get; set; }
-        public IDictionary<string, object> Properties { get; } = new Dictionary<string, object>();
+
+        public Dictionary<string, object> Properties
+        {
+            get => _properties ?? CreateDictionary();
```

Comment:
Why not `get => _properties ?? (_properties = new Dictionary<string, object>());`? I think that's a quite common pattern. Or would that make the getter to big to get inlined later?